### PR TITLE
[css-animations-2] Integrate scroll animations into animation-timeline #7579

### PR DESCRIPTION
**Pull request by fantasai**
This syntactically integrates the scroll() and view() functions into animation-timeline.

It also defines the named timeline search to look for scroll and view progress timelines before searching at-rule timelines. (This seemed the most reasonable way to integrate the search, but open to other ideas...)

### CODE REVIEW

File: css-animations-2/Overview.bs [hunk 1]
```diff
@@ -557,13 +557,27 @@ the following effects:
 ::  The animation is not associated with a [=timeline=].
 
 :   <dfn><<timeline-name>></dfn>
-::  Find the last timeline at-rule in document order with its name matching
+::  If a named [=scroll progress timeline=] or [=view progress timeline=]
+    is in scope on this element,
+    use the referenced timeline
+    as defined in [[scroll-animations-1#timeline-scope]].
+
+    Otherwise, find the last timeline at-rule in document order with its name matching
```
Comment (flackr) [inline css-animations-2/Overview.bs:565]: I believe scroll timelines were the only at-rule timelines so with those gone we can probably remove the at-rule matching.

File: css-animations-2/Overview.bs [hunk 2]
```diff
@@ -536,7 +536,7 @@ Animatable: no
 </pre>
 
 <pre class=prod>
-<dfn>&lt;single-animation-timeline></dfn> = auto | none | <<timeline-name>>
+<dfn>&lt;single-animation-timeline></dfn> = auto | none | <<timeline-name>> | <<scroll()>> | <<view()>>
```
[the remark this answers is not in the document]
Comment (fantasai) [reply css-animations-2/Overview.bs:539]: No, the angle bracket syntax `<scroll()>` is a reference to the definition of the functional notation.

File: css-animations-2/Overview.bs (hunk 1 above)
Comment (fantasai) [reply css-animations-2/Overview.bs]: Done.

File: css-animations-2/Overview.bs (hunk 2 above)
[the remark this answers is not in the document]
Comment (flackr) [reply css-animations-2/Overview.bs:539]: Thanks for explaining.

**Review by flackr — APPROVED** [review]

**Pull request merged**